jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/PushToRepositoryAction.kt

```diff
@@ -0,0 +1,365 @@
+import com.intellij.ui.components.fields.ExtendableTextField
+import com.intellij.ui.layout.GrowPolicy
+import com.intellij.ui.layout.buttonGroup
+import com.intellij.ui.layout.listCellRenderer
+import com.intellij.ui.layout.panel
+import com.intellij.ui.layout.selected
+import com.intellij.util.text.nullize
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import software.amazon.awssdk.core.exception.SdkException
+import software.amazon.awssdk.services.ecr.EcrClient
+import software.aws.toolkits.core.utils.error
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.jetbrains.core.credentials.activeRegion
+import software.aws.toolkits.jetbrains.core.explorer.ExplorerDataKeys
+import software.aws.toolkits.jetbrains.services.ecr.DockerRunConfiguration
+import software.aws.toolkits.jetbrains.services.ecr.DockerfileEcrPushRequest
+import software.aws.toolkits.jetbrains.services.ecr.EcrPushRequest
+import software.aws.toolkits.jetbrains.services.ecr.EcrRepositoryNode
+import software.aws.toolkits.jetbrains.services.ecr.EcrUtils
+import software.aws.toolkits.jetbrains.services.ecr.ImageEcrPushRequest
+import software.aws.toolkits.jetbrains.services.ecr.getDockerLogin
+import software.aws.toolkits.jetbrains.services.ecr.resources.EcrResources
+import software.aws.toolkits.jetbrains.services.ecr.resources.Repository
+import software.aws.toolkits.jetbrains.services.ecr.toLocalImageList
+import software.aws.toolkits.jetbrains.ui.ResourceSelector
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.jetbrains.utils.ui.installOnParent
+import software.aws.toolkits.jetbrains.utils.ui.selected
+import software.aws.toolkits.jetbrains.utils.ui.visibleIf
+import software.aws.toolkits.resources.message
+import software.aws.toolkits.telemetry.EcrDeploySource
+import software.aws.toolkits.telemetry.EcrTelemetry
+import software.aws.toolkits.telemetry.Result
+import javax.swing.JTextField
+import javax.swing.plaf.basic.BasicComboBoxEditor
+
+class PushToRepositoryAction :
+    DumbAwareAction(),
+    CoroutineScope by ApplicationThreadPoolScope("PushRepositoryAction") {
+    private val dockerServerRuntime: Deferred<DockerServerRuntimeInstance> =
+        async(start = CoroutineStart.LAZY) { EcrUtils.getDockerServerRuntimeInstance().runtimeInstance }
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.getRequiredData(LangDataKeys.PROJECT)
+        val client: EcrClient = project.awsClient()
+
+        val selectedRepository = e.getData(ExplorerDataKeys.SELECTED_NODES)
+            ?.mapNotNull { it as? EcrRepositoryNode }
```

Comment:
If you select 2 different types of nodes, shold we still show ECR? I think not, so maybe the take If should go first?

Reply:
sure